exporter/awskinesisexporter/internal/batch/aggregator.go

```diff
@@ -0,0 +1,75 @@
+package batch
+
+import (
+	"crypto/md5"
+
+	k "github.com/aws/aws-sdk-go/service/kinesis"
+	"google.golang.org/protobuf/proto"
+)
+
+var (
+	magicNumber           = []byte{0xF3, 0x89, 0x9A, 0xC2}
+	partitionKeyIndexSize = 8
+	maxAggregationCount   = 4294967295
+	maxRecordSize         = 1 << 20
+)
+
+type Aggregator struct {
+	buf    []*Record
+	pkeys  []string
+	nbytes int
+}
+
+func (a *Aggregator) IsRecordAggregative(data []byte, partitionKey string) bool {
+	return data != nil && len(data)+len([]byte(partitionKey)) <= maxRecordSize
```

Comment:
Just so you know, `len([]byte(nil)) == 0` so there is no need to check for nil here :)

Reply:
Furthermore, I had allowed max record size to be configurable (from memory) by the user and not allowing it to be greater than the API max, I would appreciate it if that pattern could be reused here too.